src/common/cuda/rtc/forward_functions-inl.h

```diff
@@ -697,9 +697,11 @@ __device__ inline DType log_sigmoid(const DType val) {
 template <typename DType>
 __device__ inline DType mish(const DType val) {
   if (type_util::has_double_or_integral<DType>::value) {
-    return val * ::tanh(::log(1 + ::exp(val)));
+    const auto softrelu = (val > 20) ? val : ::log(1 + ::exp(val));
+    return val * ::tanh(softrelu);
   } else {
-    return val * ::tanhf(logf(1 + expf(val)));
+    const auto softrelu = (val > 20) ? val : logf(1 + expf(val));
+    return val * ::tanhf(softrelu);
```

Comment:
Similarly here. This function body could just be
```
return val * op::tanh(op::softrelu(val));
```
right?

Reply:
Done

Reply:
Hi, @ptrendx . There are errors when using `op::softrelu(val)` or `softrelue(val)`.

```
[2021-06-12T03:43:50.408Z] E           unary_kernel_kernel.cu(1724): error: namespace "op" has no member "tanh"
[2021-06-12T03:43:50.408Z] E           
[2021-06-12T03:43:50.408Z] E           unary_kernel_kernel.cu(1724): error: namespace "op" has no member "softrelu"
```

```
[2021-06-12T05:40:39.340Z] E           FusedKernel_mish_elemwise_mul__kernel.cu(1724): error: identifier "softrelu" is undefined
[2021-06-12T05:40:39.340Z] E                     detected during instantiation of "DType op::mish(DType) [with DType=DType_input_0]" 
[2021-06-12T05:40:39.340Z] E           (3123): here
```

Reply:
That is because this function is defined "too early" in this file - e.g. softrelu function is defined just after your function (tanh is defined later in line 780). You need to move your function after those for the compiler to see them.

All of the backward functions are defined after the forward functions so that should just work if you use the `op::softrelu` and `op::tanh` there.

Reply:
fixed